src/lib.rs

```diff
@@ -1,4 +1,5 @@
 #![deny(clippy::all)]
+#![deny(unused_results)]
```

Comment:
Oh, this. But why? Did you run into some LLVM stuff that's supposed to be #[must_use]? If so let's annotate that instead?

Reply:
Nah, this doesn't really belong in this PR specifically. It's a useful lint that avoids things like partial reads and writes which report the number of bytes and folks often forget to check; see `src/linker.rs` where I changed `read` to `read_exact` because the length read wasn't being checked -- this lint surfaced that.

I'll move this to another PR so we can discuss separately from this change.